File: app/lib/api/courses_api.dart
import '../api/api_client.dart';
import '../models/enrolled_student.dart';

class CoursesApi {
  CoursesApi(this._client);

  final ApiClient _client;

  /// `GET /api/courses/:courseCode/students` — enrolled students for one course.
  /// Pass [semesterCode] (e.g. `2502`) for a historical offering roster.
  Future<CourseRoster> fetchCourseStudents(String courseCode, {String? semesterCode}) async {
    final code = courseCode.trim().toUpperCase();
    final sem = semesterCode?.trim();
    final data = await _client.requestJson(
      '/api/courses/$code/students',
      query: sem != null && sem.isNotEmpty ? {'semester': sem} : null,
    );
    return CourseRoster.fromJson(data);
  }
}

File: src/utils/coursesApi.js
import { apiFetch } from '../auth/AuthContext';

export async function fetchCourseStudents(courseCode, semesterCode) {
    const code = encodeURIComponent((courseCode || '').trim().toUpperCase());
    const sem = (semesterCode || '').trim();
    const qs = sem ? `?semester=${encodeURIComponent(sem)}` : '';
    const res = await apiFetch(`/api/courses/${code}/students${qs}`);
    if (!res.ok) throw new Error('roster_load_failed');
    const data = await res.json();
    return {
        courseCode: data.courseCode || code,
        count: data.count ?? (Array.isArray(data.students) ? data.students.length : 0),
        students: Array.isArray(data.students) ? data.students : [],
    };
}
